lib/datadog/core/workers/interval_loop.rb

```diff
@@ -89,6 +89,8 @@ def mutex
 
         def perform_loop
           mutex.synchronize do
+            return unless run_loop? # Return if thread was stopped before it even started
```

Comment:
Depending on test order, we can try to stop the main worker thread before it even started.

This doesn't break anything, but makes us wait the full `#stop` timeout, which is 1 second.

Reply:
I don't quite understand this change -- as far as I can see, `run_loop?` is only set to `true` on line 94, so how will it ever run after this change?

(I suspect the many broken specs are due to this change)

Reply:
Very correct! And CI agrees with you.

I've made changes to ensure we only return this early if `@run_loop` is explicitly set to `false`, and not uninitialized at this point.
It is only ever `@run_loop == false` if the worker has been explicitly stopped.

After this change, tests now pass 🙏